verilog/formatting/formatter.cc

```diff
@@ -230,15 +230,265 @@ static void DeterminePartitionExpansion(partition_node_type* node,
+// It returns fitting results by fitted_lines reference and wrapping decision by
+// return value.
+//
+static bool AppendFittingSubpartitions(
+    std::vector<UnwrappedLine>& fitted_lines,
+    const std::vector<UnwrappedLine>& expanded_lines,
+    const FormatStyle& style,
+    bool wrap_second_line) {
+  bool wrapped_second_line = wrap_second_line;
+
+  if (expanded_lines.size() == 0) {
+    fitted_lines.clear();
+    return wrapped_second_line;
+  } else if (expanded_lines.size() == 1) {
+    fitted_lines.push_back(expanded_lines[0]);
+    return wrapped_second_line;
+  } /* else .size() >= 2 */
+
+  // Start current_line at the begging
+  UnwrappedLine current_line(expanded_lines[0].IndentationSpaces(),
+                             expanded_lines[0].TokensRange().begin());
+  int indent;
+
+  // Wraps or not second line depending on wrap_second_line argument and fitting result
+  current_line.SpanUpToToken(expanded_lines[1].TokensRange().end());
+  if (wrap_second_line || (verible::FitsOnLine(current_line, style).fits == false)) {
+    // Forced to wrap
+
+    // Push first line
+    fitted_lines.push_back(expanded_lines[0]);
+
+    // Calculate indenting. Increase current indentation by style-defined wrapping spacing
+    indent = style.wrap_spaces + expanded_lines[0].IndentationSpaces();
+
+    // Start current line right after first line
+    current_line = UnwrappedLine(indent, expanded_lines[0].TokensRange().end());
+
+    // Ensure that current line spans up to end of second line
+    current_line.SpanUpToToken(expanded_lines[1].TokensRange().end());
+
+    // Signalize that first line was wrapped
+    wrapped_second_line = true;
+  } else {
+    // Don't have to wrap second line
+
+    // Compute new indentation level based on first line length
+    indent = verible::FitsOnLine(expanded_lines[0], style).final_column;
+
+    // current_line already spans up to end of second line
+  }
+
+  for (size_t idx = 2 ; idx < expanded_lines.size() ; ++idx) {
+    // Try to append this line
+    UnwrappedLine appendable_line(current_line);
+    appendable_line.SpanUpToToken(expanded_lines[idx].TokensRange().end());
+
+    // Test if it fits
+    if (verible::FitsOnLine(appendable_line, style).fits) {
+      // Fits, span up current_line up to the end of this line
+      current_line.SpanUpToToken(expanded_lines[idx].TokensRange().end());
+    } else {
+      // When concurrent lines exceed column limit it might happen that
+      // current_line will be empty. This if-statemnt handles this situation
+      if (current_line.Size()) {
+        // Doesn't fit, push back last fitting line
+        fitted_lines.push_back(current_line);
+
+        // And start current_line at the beginning of current not-fitting line
+        // At this moment current_line is empty
+        current_line = UnwrappedLine(indent, expanded_lines[idx].TokensRange().begin());
+      } else {
+        // Handle empty current_line case
+
```

Comment:
Just looking for clarification:
Is the intent here to let the non-fitting line go through the usual line-wrap optimization later?  If so, indicate this in a brief comment.

Reply:
Done